gtk4/src/tree_iter.rs

```diff
@@ -0,0 +1,87 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::ffi::gpointer;
+use glib::translate::*;
+
+use std::{mem, ptr};
+
+#[repr(transparent)]
+#[doc(alias = "GtkTreeIter")]
+pub struct TreeIter(ffi::GtkTreeIter);
```

Comment:
As long as you don't expose pointers in a "safe" way, it should be fine to keep the iterator `Copy` and `Clone` (and ideally also derive `Debug`).

The current implementation is also `Clone`, just that it unnecessarily uses heap allocations (which is something you fix here).